components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/handler/metrichandler/prometheus/reporter/PrometheusReporter.java

```diff
@@ -0,0 +1,266 @@
+
+        counter.labels(value).inc();
+    }
+
+    @Override
+    public void decrementCount(String metricName, Map<String, String> properties) {
+        // decrementCount() is not necessary to be implemented for the Prometheus Reporter
+        // as Gauge is used in Prometheus for the metrics that can both increment and decrement value.
+    }
+
+    @Override
+    public Object getTimer(String metricName, Map<String, String[]> properties) {
+        String[] value = properties.get(metricName);
+        Histogram timer = (Histogram) metricMap.get(metricName);
+
+        return timer.labels(value).startTimer();
+    }
+
+    @Override
+    public void observeTime(Object timer) {
+        Histogram.Timer histogramTimers = (Histogram.Timer) timer;
+        histogramTimers.observeDuration();
+    }
+
+    @Override
+    public void serverUp(String host, String port, String javaHome, String javaVersion) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVER_UP);
+        gauge.labels(host, port, javaHome, javaVersion).setToCurrentTime();
+    }
+
+    @Override
+    public void serverDown(String host, String port, String javaHome, String javaVersion) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVER_UP);
+        gauge.labels(host, port, javaHome, javaVersion).set(0);
+    }
+
+    @Override
+    public void serviceUp(String serviceName, String serviceType) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVICE_UP);
+        gauge.labels(serviceName, serviceType).setToCurrentTime();
+    }
+
+    @Override
+    public void serviceDown(String serviceName, String serviceType) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVICE_UP);
+        gauge.labels(serviceName, serviceType).set(0);
+    }
+
+    @Override
+    public void initMetrics() {
+        PrometheusMetricCreatorUtils.createProxyServiceMetric();
+        PrometheusMetricCreatorUtils.createAPIServiceMetric();
+        PrometheusMetricCreatorUtils.createInboundEndpointMetric();
+        PrometheusMetricCreatorUtils.createProxyServiceErrorMetric();
+        PrometheusMetricCreatorUtils.createApiErrorMetric();
+        PrometheusMetricCreatorUtils.createInboundEndpointErrorMetric();
+
+        PrometheusMetricCreatorUtils.createServerUpMetrics();
+        PrometheusMetricCreatorUtils.createServiceUpMetrics();
+    }
+
+    enum SERVICE {
+        PROXY,
+        API,
+        INBOUND_ENDPOINT
+    }
+
+    /**
+     * Load the user defined Histogram bucket upper limits configurations from the
+     * deployment.toml file else assign the default bucket configuration values.  .
+     *
+     * @param configs The map of configs defined for Histogram bucket upper limits
+     *                in the deployment.toml file
+     */
+    private void createBuckets(Map<String, Object> configs) {
+        proxyLatencyBuckets = new double[]{0.19, 0.20, 0.25, 0.30, 0.35, 0.40, 0.50, 0.60, 1, 5};
```

Comment:
Move these three lines to createMetrics method

Reply:
Fixed